### PR TITLE
support for data dependent schema keywords

### PR DESCRIPTION
**Pull request by PaulBredl**
Support for
- if then else
- dependent required
- dependent schemas

Remarks:
- if inside allOfs don't work well (only first if is considered). This is not due to my implementation, but due to the mergeAllOf library not supporting merging ifs. In fact, this library basically only supports merging properties and items, which is a pity. It is definitely required that we implement custom resolvers for the merge-all-of (but not necessarily scope of this project)

@Logende I added some comments at which places we could add things for oneOf/anyOf

How to test:
- use Default schema
- set isMarried to true -> spouseName property shows up (if then else)
- define a nickname -> preferredNickname property shows up (dependentSchemas)
- set the city name in the adress -> zipCode becomes required (dependentRequired)

### CODE REVIEW

File: frontend/config-assistant/src/store/sessionStore.ts [hunk 1]
```diff
@@ -157,6 +161,14 @@ export const useSessionStore = defineStore('commonStore', () => {
 
   const schemaAtCurrentPath: Ref<JsonSchema> = computed(() => schemaAtPath(currentPath.value));
 
+  const effectiveSchemaAtCurrentPath: Ref<EffectiveSchema> = computed(() =>
```
Comment (PaulBredl) [inline frontend/config-assistant/src/store/sessionStore.ts:164]: Ah shit this doesn't work at deeper levels

Comment (Logende) [reply frontend/config-assistant/src/store/sessionStore.ts:183]: okay but what should we do about it then? Do you see changing this as within the scope of this PR?

Comment (PaulBredl) [reply frontend/config-assistant/src/store/sessionStore.ts:183]: i just forgot to push

**Comment by Logende** [comment]
Wow, really cool :) Good job!

**Review by Logende — APPROVED** [review]

**Pull request merged**